libfreerdp/core/gateway/arm.c

```diff
@@ -699,54 +704,59 @@ static BOOL arm_fill_gateway_parameters(rdpArm* arm, const char* message, size_t
 	BYTE* authBlob = NULL;
 	rdpCertificate* redirectedServerCert = NULL;
 
-	do
+	if (freerdp_settings_get_string(settings, FreeRDP_Password))
 	{
-		/* redirectedAuthGuid */
-		const cJSON* redirectedAuthGuidNode =
-		    cJSON_GetObjectItemCaseSensitive(json, "redirectedAuthGuid");
-		if (!redirectedAuthGuidNode || !cJSON_IsString(redirectedAuthGuidNode))
-			break;
-
-		char* redirectedAuthGuid = cJSON_GetStringValue(redirectedAuthGuidNode);
-		if (!redirectedAuthGuid)
-			break;
-
-		size_t wGUID_len = 0;
-		wGUID = ConvertUtf8ToWCharAlloc(redirectedAuthGuid, &wGUID_len);
-		if (!wGUID)
+		/* note: we retrieve some more fields for RDSTLS only if we have a password provided by the
+		 * user, otherwise these are useless: we will not be able to do RDSTLS
+		 */
+		do
```

Comment:
maybe split this out to a `static` function? the `do {} while(FALSE)` grouping is probably easier to implement that way